server/worker/worker.go

```diff
@@ -0,0 +1,265 @@
+package worker
+
+import (
+	"encoding/json"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"unicode/utf8"
+)
+
+type Dashboard struct {
+	Uid   string
+	Data  map[string]interface{}
+	Title string
+}
+
+type DashboardsResponse struct {
+	Uid string `json:"uid"`
+}
+
+type DashboardResponse struct {
+	Dashboard map[string]interface{} `json:"dashboard"`
+}
+
+type DashboardUpdateResponse struct {
+	FolderUid string `json:"folderUid"`
+	Id        uint64 `json:"id"`
+	Slug      string `json:"slug"`
+	Status    string `json:"status"`
+	Uid       string `json:"uid"`
+	Url       string `json:"url"`
+	Version   uint64 `json:"version"`
+}
+
+func (api *Client) AddAnalyticsToDashboards() {
+	res, err := api.Get("/api/search")
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "AddAnalyticsToDashboards - Failed to get dashboard data - /api/search",
+			"error", err,
+		)
+
+		return
+	}
+
+	var response []DashboardsResponse
+	err = json.Unmarshal(res, &response)
+	if err != nil {
+		level.Info(api.Logger).Log(
+			"status", "error",
+			"message", "AddAnalyticsToDashboards - Failed to parse JSON response",
+			"error", err,
+		)
+
+		return
+	}
+
+	var dashboardsToUpdate []Dashboard
+	for _, dashboardEntry := range response {
+		hasAnalyticsPanel := false
+		rawDashboardData := api.GetDashboard(dashboardEntry.Uid)
+		dashboardData := getTypedDashboardData(rawDashboardData)
+		panels := getTypedPanelsData(dashboardData)
+
+		largestPanelId := 0
+		largestPanelId, hasAnalyticsPanel = checkAnalyticsPanelExistence(panels, largestPanelId, hasAnalyticsPanel, api.Logger)
+
+		title, ok := dashboardData["title"]
+		if !hasAnalyticsPanel && ok {
+			newAnalyticsPanel := createAnalyticsPanelData(largestPanelId+1, api.AnalyticsUrl)
+			panels = append([]interface{}{newAnalyticsPanel}, panels...)
+			dashboardData["panels"] = panels
+			rawDashboardData.Data["dashboard"] = dashboardData
+
+			dashboardsToUpdate = append(dashboardsToUpdate, Dashboard{
+				Uid:   dashboardEntry.Uid,
+				Data:  rawDashboardData.Data,
+				Title: title.(string),
+			})
+		}
+	}
+
+	api.updateDashboards(dashboardsToUpdate)
+}
+
+func (api *Client) updateDashboards(dashboards []Dashboard) {
+	hasFilter := false
+
+	if utf8.RuneCountInString(api.Filter) > 0 {
+		hasFilter = true
+	}
+
+	for _, dashboard := range dashboards {
+		api.updateDashboard(dashboard, hasFilter)
+	}
+}
+
+func (api *Client) updateDashboard(dashboard Dashboard, hasFilter bool) {
+	if hasFilter && dashboard.Title != api.Filter {
```

Comment:
i'd move this if statement around this function call, so the function signature would make more sense (`hasFilter` currently isn't an input to updating the dashboard, but decides if dashboard is updated at all)